python/ccxt/async_support/base/exchange.py

```diff
@@ -142,6 +148,7 @@ async def fetch(self, url, method='GET', headers=None, body=None):
                 raise NotSupported(self.id + ' - to use SOCKS proxy with ccxt, you need "aiohttp_socks" module that can be installed by "pip install aiohttp_socks"')
             # Create our SSL context object with our CA cert file
             context = ssl.create_default_context(cafile=self.cafile) if self.verify else self.verify
+            self.open()  # ensure `asyncio_loop` is set
```

Comment:
I've moved `self.open` from few lines down to here. otherwise, if `self.open()` is left where it was, and if you run `examples/python/proxy-usage.py` in debugger, and place breakpoint at `python\ccxt\async_support\base\ws\fast_client.py`th 35 line, where it says `self.asyncio_loop.call_soon(handler)` , the thing is that, `self.asyncio_loop` is None by that time, and also the exception is somewhat muted and nowhere throw (have spent some while to debug that particular problem).
the thing is that, as I see, the `open` method (https://github.com/ccxt/ccxt/blob/8c4c067c05f400d519aabd399da3f6bb062883bd/python/ccxt/async_support/base/exchange.py#L104) sets `asyncio_loop`.
 `open` happens inside `fetch` https://github.com/ccxt/ccxt/blob/8c4c067c05f400d519aabd399da3f6bb062883bd/python/ccxt/async_support/base/exchange.py#L392 and in `watch` (https://github.com/ccxt/ccxt/blob/8c4c067c05f400d519aabd399da3f6bb062883bd/python/ccxt/async_support/base/exchange.py#L392)  (also in `__aenter__`, but it's not relevant atm). so, all before that `asyncio_loop` is `None`. 
To add further note, here  : https://github.com/ccxt/ccxt/blob/8c4c067c05f400d519aabd399da3f6bb062883bd/python/ccxt/async_support/base/exchange.py#L381 client is called `self.client(url)` and few lines above at 365th line, the `asyncio_loop` is being referred, but it is not set by that time, as `watch` method, atm, calls `self.open()` after `self.client(url)`. 

wdyt:
1) should we move `self.open` a bit upper in `fetch & watch` methods? as I tested, doing that fixes the problem. wdyt?
2) should we change smth, so that exception I've mentioned, was not lost?

Reply:
@carlosmiei I've moved comment from discord to here ^